jsonrpc/types.go

```diff
@@ -108,28 +109,69 @@ func toTransaction(
 	return res
 }
 
+type header struct {
+	ParentHash      types.Hash  `json:"parentHash"`
+	Sha3Uncles      types.Hash  `json:"sha3Uncles"`
+	Miner           argBytes    `json:"miner"`
+	StateRoot       types.Hash  `json:"stateRoot"`
+	TxRoot          types.Hash  `json:"transactionsRoot"`
+	ReceiptsRoot    types.Hash  `json:"receiptsRoot"`
+	LogsBloom       types.Bloom `json:"logsBloom"`
+	Difficulty      argUint64   `json:"difficulty"`
+	TotalDifficulty argUint64   `json:"totalDifficulty"`
+	Number          argUint64   `json:"number"`
+	GasLimit        argUint64   `json:"gasLimit"`
+	GasUsed         argUint64   `json:"gasUsed"`
+	Timestamp       argUint64   `json:"timestamp"`
+	ExtraData       argBytes    `json:"extraData"`
+	MixHash         types.Hash  `json:"mixHash"`
+	Nonce           types.Nonce `json:"nonce"`
+	Hash            types.Hash  `json:"hash"`
+	BaseFee         argUint64   `json:"baseFeePerGas,omitempty"`
+}
+
+type accessListResult struct {
+	Accesslist types.TxAccessList `json:"accessList"`
+	GasUsed    argUint64          `json:"gasUsed"`
+}
+
 type block struct {
-	ParentHash      types.Hash          `json:"parentHash"`
-	Sha3Uncles      types.Hash          `json:"sha3Uncles"`
-	Miner           argBytes            `json:"miner"`
-	StateRoot       types.Hash          `json:"stateRoot"`
-	TxRoot          types.Hash          `json:"transactionsRoot"`
-	ReceiptsRoot    types.Hash          `json:"receiptsRoot"`
-	LogsBloom       types.Bloom         `json:"logsBloom"`
-	Difficulty      argUint64           `json:"difficulty"`
-	TotalDifficulty argUint64           `json:"totalDifficulty"`
-	Size            argUint64           `json:"size"`
-	Number          argUint64           `json:"number"`
-	GasLimit        argUint64           `json:"gasLimit"`
-	GasUsed         argUint64           `json:"gasUsed"`
-	Timestamp       argUint64           `json:"timestamp"`
-	ExtraData       argBytes            `json:"extraData"`
-	MixHash         types.Hash          `json:"mixHash"`
-	Nonce           types.Nonce         `json:"nonce"`
-	Hash            types.Hash          `json:"hash"`
-	Transactions    []transactionOrHash `json:"transactions"`
-	Uncles          []types.Hash        `json:"uncles"`
-	BaseFee         argUint64           `json:"baseFeePerGas,omitempty"`
+	header
+	Size         argUint64           `json:"size"`
+	Transactions []transactionOrHash `json:"transactions"`
+	Uncles       []types.Hash        `json:"uncles"`
+}
+
+func toAccessList(aList *runtime.AccessList) types.TxAccessList {
+	convertMapToStorageKeys := func(m map[types.Hash]struct{}) []types.Hash {
+		r := make([]types.Hash, len(m))
+		index := 0
+
+		for key := range m {
+			r[index] = key
+			index++
+		}
+
+		return r
+	}
+
+	if aList == nil {
+		return make(types.TxAccessList, 0)
+	}
+
+	result := make(types.TxAccessList, len(*aList))
+	idx := 0
```

Comment:
Consider this change. It would allocate capacity of the slice. And then within the for loop, you can use append instead of indexing and remove the `idx` variable.

```suggestion
	result := make(types.TxAccessList, 0, len(aList))
```

```go
for address, item := range aList {
    result = append(result, types.AccessTuple{
			Address:     address,
			StorageKeys: convertMapToStorageKeys(item),
		})
}
```